WooCommerce/Classes/POS/Presentation/Reusable Views/POSSendReceiptModalView.swift

```diff
@@ -0,0 +1,103 @@
+import SwiftUI
+
+struct POSSendReceiptModalView: View {
+    @EnvironmentObject private var posModel: PointOfSaleAggregateModel
```

Comment:
Totally fine and what it's meant for, but there are still arguments where we don't really need to observe state, to just pass the bits of the posModel that we actually do need in. You could replace this with a property to hold the send receipt function.

The benefit is that it becomes usable outside POS, which seems unlikely, but it's also simpler for creating the preview.

Reply:
So just send the email back as a callback, and then handle both the call to send the receipt and dismiss from the TotalsView? Eg:
```

// POSSendReceiptModalView:

var completion: (String) -> ()

Button(action: {
  Task { @MainActor in
    inputCallback(textFieldInput)
  }
}

// TotalsView

.sheet(isPresented: $isShowingSendReceiptModal) {
  POSSendReceiptModalView(inputCallback: { email in
    Task { await posModel.sendReceipt(to: email)  } 
  })
}
```

Reply:
Yep, but I'd call it `sendReceipt`, not `completion`, on the modal too.

Reply:
Gotcha, done: https://github.com/woocommerce/woocommerce-ios/pull/14481/commits/3169cbb7a50eb4e17a8d33dd50b14f11765a7d51